Sources/SwiftCrossUI/Views/DatePicker.swift

```diff
@@ -0,0 +1,152 @@
+    ///   - range: The range of dates to display. The backend takes this as a hint but it is not
+    ///     necessarily enforced. As such this parameter should be treated as an aid to validation
+    ///     rather than a replacement for it.
+    ///   - displayedComponents: What parts of the date/time to display in the input.
+    ///   - label: The view to be shown next to the date input.
+    public nonisolated init(
+        selection: Binding<Date>,
+        range: ClosedRange<Date> = Date.distantPast...Date.distantFuture,
+        displayedComponents: DatePickerComponents = [.hourAndMinute, .date],
+        @ViewBuilder label: () -> Label
+    ) {
+        self.label = label()
+        self.selection = selection
+        self.range = range
+        self.components = displayedComponents
+    }
+
+    /// Displays a date input.
+    /// - Parameters:
+    ///   - label: The text to be shown next to the date input.
+    ///   - selection: The currently-selected date.
+    ///   - range: The range of dates to display. The backend takes this as a hint but it is not
+    ///     necessarily enforced. As such this parameter should be treated as an aid to validation
+    ///     rather than a replacement for it.
+    ///   - displayedComponents: What parts of the date/time to display in the input.
+    public nonisolated init(
+        _ label: String,
+        selection: Binding<Date>,
+        range: ClosedRange<Date> = Date.distantPast...Date.distantFuture,
+        displayedComponents: DatePickerComponents = [.hourAndMinute, .date]
+    ) where Label == Text {
+        self.label = Text(label)
+        self.selection = selection
+        self.range = range
+        self.components = displayedComponents
+    }
+
+    public typealias Components = DatePickerComponents
+}
+
+@available(tvOS, unavailable)
+extension DatePicker: View {
+    public var body: some View {
+        HStack {
+            label
+
+            DatePickerImplementation(selection: selection, range: range, components: components)
+        }
+    }
+}
+
+@available(tvOS, unavailable)
+internal struct DatePickerImplementation: ElementaryView {
+    @Binding private var selection: Date
+    private var range: ClosedRange<Date>
+    private var components: DatePickerComponents
+
+    init(selection: Binding<Date>, range: ClosedRange<Date>, components: DatePickerComponents) {
+        self._selection = selection
+        self.range = range
+        self.components = components
+    }
+
+    let body = EmptyView()
+
+    func asWidget<Backend: AppBackend>(backend: Backend) -> Backend.Widget {
+        backend.createDatePicker()
+    }
+
+    func update<Backend: AppBackend>(
+        _ widget: Backend.Widget,
+        proposedSize: SIMD2<Int>,
+        environment: EnvironmentValues,
+        backend: Backend,
+        dryRun: Bool
+    ) -> ViewUpdateResult {
+        #if os(tvOS)
+            preconditionFailure()
+        #else
```

Comment:
With my comment about making `updateDatePicker` available on tvOS, I think this conditional compilation should be made redundant.

Reply:
I removed it elsewhere but not here, oops :sweat_smile: fixed